src/python/gem5/components/boards/abstract_board.py

```diff
@@ -369,8 +401,8 @@ def _connect_things(self) -> None:
         self.get_memory().incorporate_memory(self)
 
         # Incorporate the cache hierarchy for the motherboard.
-        if self.get_cache_hierarchy():
-            self.get_cache_hierarchy().incorporate_cache(self)
+        if cache_hierarchy := self.get_cache_hierarchy():
```

Comment:
I don't like the idea of the walrus operator... call me old fashioned...

IMO, the tradeoff of slightly harder to read to save 9 characters isn't worth it.

Reply:
Although I disagree, that's fair.
The alternative is to throw `self.get_cache_hierarchy()` into a temp variable and then check for None, the current approach only works as expected if the function is a pure function (no side effects), and although that tends to be true it can't be guaranteed and may change in the future.

Something like

```python
cache_hierarchy = self.get_cache_hierarchy()
if cache_hierarchy:
    ...
```